flytekit/remote/remote.py

```diff
@@ -1380,11 +1382,11 @@ def _assign_inputs_and_outputs(
         """Helper for assigning synced inputs and outputs to an execution object."""
         with self.remote_context():
```

Comment:
we don't really need to add this extra context manager now that we're specifying the context in the constructor of flyteremote, right? This applies to the other uses of `FlyteContext` in this file as well.

Reply:
hmm.  i think this line is probably not needed, but other usages of the remote_context function are still required right?  For things like pulling down data and stuff, we need to use the remote instance's data config.

Reply:
yes, for example in places like [1](https://github.com/flyteorg/flytekit/blob/master/flytekit/remote/remote.py#L429), [2](https://github.com/flyteorg/flytekit/blob/master/flytekit/remote/remote.py#L498), and [3](https://github.com/flyteorg/flytekit/blob/master/flytekit/remote/remote.py#L1002), right?